### PR TITLE
Mark win-64 py39 as broken due to issues with entrypoints

### PR DESCRIPTION
**Pull request by mariusvniekerk**
Guidelines for marking packages as broken:

* We prefer to patch the repo data (see [here](https://github.com/conda-forge/conda-forge-repodata-patches-feedstock))
  instead of marking packages as broken. This alternative workflow makes environments more reproducible.
* Packages with requirements/metadata that are too strict but otherwise work are
  not technically broken and should not be marked as such.
* Packages with missing metadata can be marked as broken on a temporary basis
  but should be patched in the repo data and be marked unbroken later.
* In some cases where the number of users of a package is small or it is used by
  the maintainers only, we can allow packages to be marked broken more liberally.
* We (`conda-forge/core`) try to make a decision on these requests within 24 hours.

Checklist:

* [ ] Make sure your package is in the right spot (`broken/*` for adding the
  `broken` label, `not_broken/*` for removing the `broken` label)
* [ ] Added a description of the problem with the package in the PR description.
* [ ] Added links to any relevant issues/PRs in the PR description.
* [ ] Pinged the team for the package for their input.

### CODE REVIEW

**Review by isuruf — CHANGES_REQUESTED** [review]
It's not a python issue. It's a conda-build issue

**Comment by mariusvniekerk** [comment]
aren't those exit codes real?

**Comment by jakirkham** [comment]
Do we have a sense of what `conda-build` is doing wrong w.r.t. Python 3.9? Is it possible that Python 3.9 made some changes to how Windows entry points are created (nothing sticks out to me from the release notes, but could be buried)?

**Comment by isuruf** [comment]
I take that back. I tried an older conda-build/conda versions and it worked two times, but running more will reproduce.

**Comment by mbargull** [comment]
> Is it possible that Python 3.9 made some changes to how Windows entry points are created

`pip` or `conda` creates `console_scripts` entry points. But why would it be entry point-related? In https://github.com/conda-forge/setuptools-feedstock/issues/178 it probably didn't run any of those.

**Comment by mariusvniekerk** [comment]
So it seems to be we're getting a segfault (access violation)  when tearing down the python interpreter.

**Comment by mariusvniekerk** [comment]
I don't quite know the right sequence of magic flags to get a meaningful stack dump at the c level in windows

Current as far as

```
"C:\\Program Files (x86)\\Windows Kits\\10\\Debuggers\\x64\\cdb.exe" python.exe "%PREFIX%\\Library\\bin\\g-ir-scanner" --help
```

but that seems to just start the application not run it?

**Comment by isuruf** [comment]
Looks like the last conda package on win-64 is to blame. I manually reverted the patch in https://github.com/conda-forge/conda-feedstock/pull/128/files, but segfaults are there

**Comment by mbargull** [comment]
> So it seems to be we're getting a segfault (access violation) when tearing down the python interpreter.

(Just to give context, because the issue discussion is scattered in different places, @mariusvniekerk is referring to https://gitter.im/conda-forge/regro-cf-autotick-bot?at=5f80871a99e1ab4dd1ebce75, i.e., he's debugging at https://dev.azure.com/conda-forge/feedstock-builds/_build?definitionId=379)

**Comment by mbargull** [comment]
> , but segfaults are there

What do you mean with this?

**Comment by isuruf** [comment]
Manually reverting the [patch](https://github.com/conda-forge/conda-feedstock/pull/128/files) in conda `_2` doesn't fix this problem. But reverting to conda `_1` does

File: broken/py39-win-and-pre.txt [hunk 1]
```diff
@@ -0,0 +1,5 @@
+win-64/python-3.9.0-h60c2a47_1_cpython.tar.bz2
+win-64/python-3.9.0-h60c2a47_2_cpython.tar.bz2
+win-64/python-3.9.0-h60c2a47_0_cpython.tar.bz2
+osx-64/python-3.9.0-hcfdab8c_0_cpython.tar.bz2
+linux-64/python-3.9.0-h852b56e_0_cpython.tar.bz2
```
Comment (beckermr) [inline broken/py39-win-and-pre.txt:5]:
```suggestion
```

These are already marked as broken. That is why the CI is failing.

**Comment by mariusvniekerk** [comment]
so should we yank a version of conda?

**Comment by mbargull** [comment]
I don't understand why/how `conda` would influence segfaults in `python`, though.

**Comment by mariusvniekerk** [comment]
```
(1bc8.19c): Access violation - code c0000005 (first chance)
First chance exceptions are reported before any exception handling.
This exception may be expected and handled.
*** WARNING: Unable to verify checksum for %PREFIX%\python39.dll
python39!gc_reset_refs [inlined in python39!deduce_unreachable+0x23]:
00007ffa`1e1affd3 4883630801      and     qword ptr [rbx+8],1 ds:00000000`00000008=????????????????
```

**Comment by mariusvniekerk** [comment]
presumably i need some symbols

**Comment by isuruf** [comment]
Nvm, even though I can't reproduce with `conda` `*_1` builds, CI could reproduce.

**Comment by mariusvniekerk** [comment]
so do we have symbols for python?

**Comment by mariusvniekerk** [comment]
from that call it looks like its trying to reset the gc after it had already been deallocated?

https://github.com/python/cpython/blob/master/Modules/gcmodule.c#L106

**Comment by mariusvniekerk** [comment]
This https://bugs.python.org/issue38379 might be relevant

**Comment by mariusvniekerk** [comment]
Okay got a backtrace out of it

```
2020-10-09T19:43:03.3379467Z (1de0.1904): Access violation - code c0000005 (first chance)
2020-10-09T19:43:03.3379839Z First chance exceptions are reported before any exception handling.
2020-10-09T19:43:03.3380357Z This exception may be expected and handled.
2020-10-09T19:43:03.4822636Z *** WARNING: Unable to verify checksum for %PREFIX%\python39.dll
2020-10-09T19:43:03.5286260Z Child-SP          RetAddr           : Args to Child                                                           : Call Site
2020-10-09T19:43:03.5288293Z (Inline Function) --------`-------- : --------`-------- --------`-------- --------`-------- --------`-------- : python39!gc_reset_refs (Inline Function @ 00007ffe`818dffd3)
2020-10-09T19:43:03.5289688Z (Inline Function) --------`-------- : --------`-------- --------`-------- --------`-------- --------`-------- : python39!update_refs+0xe (Inline Function @ 00007ffe`818dffd3)
2020-10-09T19:43:03.5290640Z 000000ce`8f7ef0e0 00007ffe`818e037a : 000002ae`ac020cc0 000002ae`ac05bf00 00007ffe`81cd8db0 00007ffe`81850000 : python39!deduce_unreachable+0x23
2020-10-09T19:43:03.5291464Z 000000ce`8f7ef110 00007ffe`81ab5b15 : 000002ae`00000001 000002ae`ac020a10 000002ae`b22ffe00 00000000`00000121 : python39!collect+0x20a
2020-10-09T19:43:03.5292418Z (Inline Function) --------`-------- : --------`-------- --------`-------- --------`-------- --------`-------- : python39!_PyGC_CollectNoFail+0x35 (Inline Function @ 00007ffe`81ab5b15)
2020-10-09T19:43:03.5304112Z 000000ce`8f7ef290 00007ffe`81ad1122 : 000002ae`ac05bf00 ffffffff`00000002 000002ae`ac804f00 ffffffff`ffffffff : python39!_PyImport_Cleanup+0xa15
2020-10-09T19:43:03.5305725Z 000000ce`8f7ef350 00007ffe`81ad339d : 00000000`00000000 00000000`00000000 000000ce`8f7ef449 ffffffff`ffffffff : python39!Py_FinalizeEx+0x212
2020-10-09T19:43:03.5307275Z 000000ce`8f7ef3b0 00007ffe`81af64af : 00000000`00000000 000000ce`8f7ef449 ffffffff`ffffffff 000000ce`8f7ef4b8 : python39!Py_Exit+0xd
2020-10-09T19:43:03.5308815Z (Inline Function) --------`-------- : --------`-------- --------`-------- --------`-------- --------`-------- : python39!handle_system_exit+0x4c9 (Inline Function @ 00007ffe`81af64af)
2020-10-09T19:43:03.5310409Z 000000ce`8f7ef3e0 00007ffe`81af5341 : 00000000`00000000 00000000`00000d61 ffffffff`ffffffff 000002ae`ae9cb910 : python39!_PyErr_PrintEx+0x4ef
2020-10-09T19:43:03.5312059Z (Inline Function) --------`-------- : --------`-------- --------`-------- --------`-------- --------`-------- : python39!PyErr_PrintEx+0x11 (Inline Function @ 00007ffe`81af5341)
2020-10-09T19:43:03.5318840Z (Inline Function) --------`-------- : --------`-------- --------`-------- --------`-------- --------`-------- : python39!PyErr_Print+0x11 (Inline Function @ 00007ffe`81af5341)
2020-10-09T19:43:03.5325704Z 000000ce`8f7ef4b0 00007ffe`818ec0ad : 000000ce`8f7ef640 000002ae`aea21a30 00000000`00000001 000000ce`8f7ef640 : python39!PyRun_SimpleFileExFlags+0x351
2020-10-09T19:43:03.5395119Z (Inline Function) --------`-------- : --------`-------- --------`-------- --------`-------- --------`-------- : python39!PyRun_AnyFileExFlags+0x46 (Inline Function @ 00007ffe`818ec0ad)
2020-10-09T19:43:03.5397643Z 000000ce`8f7ef540 00007ffe`818ecd6a : 000002ae`ac020a10 000002ae`ac020a10 00007ffe`81c4f880 00000000`00000001 : python39!pymain_run_file+0x2dd
2020-10-09T19:43:03.5399460Z 000000ce`8f7ef610 00007ffe`818edc52 : 000000ce`8f7ef720 000000ce`8f7ef720 00000000`00000000 00000000`00000000 : python39!pymain_run_python+0x4da
2020-10-09T19:43:03.5401143Z (Inline Function) --------`-------- : --------`-------- --------`-------- --------`-------- --------`-------- : python39!Py_RunMain+0xe (Inline Function @ 00007ffe`818edc52)
2020-10-09T19:43:03.5402876Z 000000ce`8f7ef690 00007ffe`818edcc6 : 00000000`00000000 20000400`00000000 00007ff6`96e43290 00007ff6`96e41dd9 : python39!pymain_main+0x52
2020-10-09T19:43:03.6749818Z *** WARNING: Unable to verify checksum for python.exe
2020-10-09T19:43:03.6792696Z 000000ce`8f7ef700 00007ff6`96e414f8 : 00000000`00000000 00000000`00000000 00000000`00000000 00000000`00000000 : python39!Py_Main+0x26
2020-10-09T19:43:04.1865975Z (Inline Function) --------`-------- : --------`-------- --------`-------- --------`-------- --------`-------- : python!invoke_main+0x22 (Inline Function @ 00007ff6`96e414f8)
2020-10-09T19:43:04.1867624Z 000000ce`8f7ef750 00007ffe`b4f884d4 : 00000000`00000000 00000000`00000000 00000000`00000000 00000000`00000000 : python!__scrt_common_main_seh+0x10c
2020-10-09T19:43:04.1869565Z 000000ce`8f7ef790 00007ffe`b50ce871 : 00000000`00000000 00000000`00000000 00000000`00000000 00000000`00000000 : KERNEL32!BaseThreadInitThunk+0x14
2020-10-09T19:43:04.1870749Z 000000ce`8f7ef7c0 00000000`00000000 : 00000000`00000000 00000000`00000000 00000000`00000000 00000000`00000000 : ntdll!RtlUserThreadStart+0x21
2020-10-09T19:43:04.1871604Z quit:
2020-10-09T19:43:04.1872195Z NatVis script unloaded from 'C:\Program Files (x86)\Windows Kits\10\Debuggers\x64\Visualizers\atlmfc.natvis'
2020-10-09T19:43:04.1873163Z NatVis script unloaded from 'C:\Program Files (x86)\Windows Kits\10\Debuggers\x64\Visualizers\concurrency.natvis'
2020-10-09T19:43:04.1875846Z NatVis script unloaded from 'C:\Program Files (x86)\Windows Kits\10\Debuggers\x64\Visualizers\cpp_rest.natvis'
2020-10-09T19:43:04.1880652Z NatVis script unloaded from 'C:\Program Files (x86)\Windows Kits\10\Debuggers\x64\Visualizers\stl.natvis'
2020-10-09T19:43:04.1881973Z NatVis script unloaded from 'C:\Program Files (x86)\Windows Kits\10\Debuggers\x64\Visualizers\Windows.Data.Json.natvis'
2020-10-09T19:43:04.1883333Z NatVis script unloaded from 'C:\Program Files (x86)\Windows Kits\10\Debuggers\x64\Visualizers\Windows.Devices.Geolocation.natvis'
2020-10-09T19:43:04.1890999Z NatVis script unloaded from 'C:\Program Files (x86)\Windows Kits\10\Debuggers\x64\Visualizers\Windows.Devices.Sensors.natvis'
2020-10-09T19:43:04.1893973Z NatVis script unloaded from 'C:\Program Files (x86)\Windows Kits\10\Debuggers\x64\Visualizers\Windows.Media.natvis'
2020-10-09T19:43:04.1897552Z NatVis script unloaded from 'C:\Program Files (x86)\Windows Kits\10\Debuggers\x64\Visualizers\windows.natvis'
2020-10-09T19:43:04.1908902Z NatVis script unloaded from 'C:\Program Files (x86)\Windows Kits\10\Debuggers\x64\Visualizers\winrt.natvis'
```

**Comment by mariusvniekerk** [comment]
So any thoughts on how / why @isuruf @mingwandroid ?

**Comment by mariusvniekerk** [comment]
Also maybe @jjhelmus

**Comment by mariusvniekerk** [comment]
Tests ran over here https://github.com/conda-forge/gobject-introspection-feedstock/pull/28

**Comment by mingwandroid** [comment]
No but I'll take a look for a few hours soon. I am having issues with pgo on windows python 3.9.0 though so I'll try to fix that first (for defaults for now, still cannot get CF ci to like pgo..)

Hey at least my head is on the right build space!

@mariusvniekerk did you see the debugging stuff in the recipe? Was it if any use?

**Comment by mariusvniekerk** [comment]
Haven't quite gone that far into it @mingwandroid.  The new debug stuff looks neat, we just need to build a python with it for our purposes :)

**Comment by mingwandroid** [comment]
I have used it recently but I it does usually require some minor gardening each time!

**Comment by mingwandroid** [comment]
BTW I don't know if we need a separate abi (and the _d suffixes) on windows anymore for debug builds. It is something to investigate.

If no then we're in a great position to provide debug variants of just the python package itself for when people need to debug stuff. We should probably go ahead with this on unix anyway!

**Comment by beckermr** [comment]
So @mariusvniekerk should we hold off on this PR or go ahead with it so we don't push as many bad PRs with the bot?

**Comment by mingwandroid** [comment]
I've not gotten round to this yet, may not until tomorrow now.

**Comment by mingwandroid** [comment]
Busy adding more patching love to conda-build

**Comment by mariusvniekerk** [comment]
@mingwandroid with some debug symbols i got 

```
00 000000a3`0dfe4f80 00007ffb`d9aa5c77 : 00000000`00000000 000001b0`b207e7e0 000001b0`b2081140 00007ffc`6113fcb0 : python39_d!fatal_output_debug+0x59 [\Python\pylifecycle.c @ 2146] 
01 000000a3`0dfe50f0 00007ffb`d9aa34c9 : 00007ffc`6113fcb0 00007ffc`00000000 00000000`00000000 00000000`00000000 : python39_d!fatal_error+0x1b7 [\Python\pylifecycle.c @ 2290] 
02 000000a3`0dfe5150 00007ffb`d980871f : 00007ffb`d9ca6a20 00007ffb`d9ca6b00 00000000`0000006f 00000000`0000006d : python39_d!_Py_FatalErrorFormat+0xe9 [\Python\pylifecycle.c @ 2341] 
03 000000a3`0dfe51a0 00007ffb`d98079aa : 00007ffb`d9ca6a20 00000000`0000006d 000001b0`b2a94f20 00007ffc`60ffad59 : python39_d!_PyMem_DebugCheckAddress+0x8f [\Objects\obmalloc.c @ 2386] 
04 000000a3`0dfe51f0 00007ffb`d9807b69 : 00007ffb`d9da6220 000001b0`b2a94f20 00007ffb`d9ded978 000001b0`00000000 : python39_d!_PyMem_DebugRawFree+0x4a [\Objects\obmalloc.c @ 2209] 
05 000000a3`0dfe5240 00007ffb`d980800b : 00007ffb`d9da6220 000001b0`b2a94f20 000001b0`b2081140 00000000`00000000 : python39_d!_PyMem_DebugFree+0x39 [\Objects\obmalloc.c @ 2346] 
06 000000a3`0dfe5270 00007ffb`d9a47a91 : 000001b0`b2a94f20 000001b0`b2081140 000001b0`b201c530 00007ffc`780a0000 : python39_d!PyMem_Free+0x1b [\Objects\obmalloc.c @ 630] 
07 000000a3`0dfe52a0 00007ffb`d9a7bc72 : 00007ffb`d9d24fb4 000001b0`b2a8c7a0 000001b0`b2a85640 00000000`00000000 : python39_d!_PyImport_FindSharedFuncptrWindows+0x131 [\Python\dynload_win.c @ 214] 
08 000000a3`0dfe5790 00007ffb`d9a7846a : 000001b0`b2aa0410 00000000`00000000 000001b0`ffffffff 000001b0`b2a8ecef : python39_d!_PyImport_LoadDynamicModuleWithSpec+0x1d2 [\Python\importdl.c @ 128] 
09 000000a3`0dfe58a0 00007ffb`d9a72264 : 000001b0`b2013530 000001b0`b2aa0410 00000000`00000000 000001b0`b2081140 : python39_d!_imp_create_dynamic_impl+0x30a [\Python\import.c @ 2294] 
0a 000000a3`0dfe5930 00007ffb`d97f7c7a : 000001b0`b2013530 000001b0`b2a89518 00000000`00000001 00000000`00000040 : python39_d!_imp_create_dynamic+0xa4 [\Python\clinic\import.c.h @ 330] 
0b 000000a3`0dfe5980 00007ffb`d9756d5a : 000001b0`b201c530 000001b0`b2a89518 00000000`00000001 00000000`00000000 : python39_d!cfunction_vectorcall_FASTCALL+0xea [\Objects\methodobject.c @ 424] 
0c 000000a3`0dfe59e0 00007ffb`d9757b63 : 000001b0`b201c530 000001b0`b2a89500 000001b0`b2a8ed10 000001b0`00000000 : python39_d!PyVectorcall_Call+0x22a [\Objects\call.c @ 230] 
0d 000000a3`0dfe5a80 00007ffb`d9756228 : 000001b0`b2081140 000001b0`b201c530 000001b0`b2a89500 000001b0`b2a8ed10 : python39_d!_PyObject_Call+0x143 [\Objects\call.c @ 265] 
0e 000000a3`0dfe5ae0 00007ffb`d9a139c8 : 000001b0`b201c530 000001b0`b2a89500 000001b0`b2a8ed10 000001b0`b2a82e30 : python39_d!PyObject_Call+0x48 [\Objects\call.c @ 293] 
0f 000000a3`0dfe5b20 00007ffb`d9a0e64b : 000001b0`b2a8ed10 000001b0`b2a82e30 000000a3`0dfe5c60 000001b0`b201c530 : python39_d!do_call_core+0x398 [\Python\ceval.c @ 5064] 
10 000000a3`0dfe5b60 00007ffb`d9a079fa : 000001b0`b2a43968 00000000`00000001 00000000`00000000 00000000`00000000 : python39_d!_PyEval_EvalFrameDefault+0x681b [\Python\ceval.c @ 3553] 
```

**Comment by isuruf** [comment]
If I replace just `python39.dll` (and nothing else) from python official release, the error goes away,

**Comment by isuruf** [comment]
From @mariusvniekerk's backtrace, it looks like https://github.com/conda-forge/python-feedstock/blob/master/recipe/patches/0024-Add-CONDA_DLL_SEARCH_MODIFICATION_KEEP_GIL-to-aid-de.patch is at fault

**Comment by mbargull** [comment]
Thanks for the digging @isuruf and @mariusvniekerk!

> From @mariusvniekerk's backtrace, it looks like https://github.com/conda-forge/python-feedstock/blob/master/recipe/patches/0024-Add-CONDA_DLL_SEARCH_MODIFICATION_KEEP_GIL-to-aid-de.patch is at fault

So maybe the extra
```diff
+#if !USE_UNICODE_WCHAR_CACHE
+        PyMem_Free(wpathname);
+#endif /* USE_UNICODE_WCHAR_CACHE */
```
?

**Comment by isuruf** [comment]
Yes, https://github.com/conda-forge/python-feedstock/pull/401/files

**Comment by isuruf** [comment]
Thanks @mariusvniekerk

**Pull request merged**